plugin/pkg/auth/authenticator/token/keystone/keystone.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keystone
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack"
+	"k8s.io/kubernetes/pkg/auth/user"
+	openstackcp "k8s.io/kubernetes/pkg/cloudprovider/providers/openstack"
```

Comment:
I'm going to poke around a little bit and see if we can share just the keystone part. When I was in there last, I think there was a good place to do that, but thought I could share more code the other way. But since that inits twice, we can probably back it off to just the one part.